File: static/js/calc.js
// 依照後端邏輯即時計算總花費
function calculateTotal() {
    function val(name) {
        return parseFloat(document.querySelector(`[name="${name}"]`).value) || 0;
    }
    let ticket = val('ticket');
    let hotel = val('hotel');
    let sim = val('sim');
    let tickets = val('tickets');
    let exchange = val('exchange');
    let insurance = val('insurance');
    let card = val('card');
    let currency_have = val('currency_have');
    let currency_left = val('currency_left');
    let rate = val('rate');
    let card_twd = rate ? card / rate : 0;
    let currency_spent = currency_have - currency_left;
    let currency_spent_twd = rate ? currency_spent * rate : 0;
    let total = ticket + hotel + sim + tickets + exchange + insurance + card_twd + currency_spent_twd;
    return Math.round(total * 100) / 100;
}

function updateTotalBox() {
    const box = document.getElementById('js-result-box');
    if (!box) return;
    const total = calculateTotal();
    box.innerHTML = `<i class='bi bi-cash-coin'></i> <span>總花費 (TWD)：</span> ${total}`;
}

['ticket','hotel','sim','tickets','exchange','insurance','card','currency_have','currency_left','rate'].forEach(name => {
    document.addEventListener('input', function(e) {
        if (e.target.name === name) updateTotalBox();
    });
});

window.addEventListener('DOMContentLoaded', updateTotalBox); 

File: utils/calc.py
def calculate_total(data):
    try:
        ticket = float(data.get('ticket', 0))
        hotel = float(data.get('hotel', 0))
        sim = float(data.get('sim', 0))
        tickets = float(data.get('tickets', 0))
        exchange = float(data.get('exchange', 0))
        insurance = float(data.get('insurance', 0))
        card = float(data.get('card', 0))
        currency_have = float(data.get('currency_have', 0))
        currency_left = float(data.get('currency_left', 0))
        rate = float(data.get('rate', 0))
        card_twd = card / rate if rate else 0
        currency_spent = currency_have - currency_left
        currency_spent_twd = currency_spent * rate if rate else 0
        total = ticket + hotel + sim + tickets + exchange + insurance + card_twd + currency_spent_twd
        return round(total, 2)
    except Exception:
        return 0 